test/unit/rules/no-action-on-submit-button-test.js

```diff
@@ -0,0 +1,115 @@
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+const ERROR_MESSAGE = 'A `<button>` with `type="submit"` should have no click action';
+
+generateRuleTests({
+  name: 'no-action-on-submit-button',
+
+  config: true,
+
+  good: [
+    // valid buttons with "button" type
+    '<button type="button" />',
+    '<button type="button" {{action this.handleClick"}} />',
+    '<button type="button" {{action this.handleClick" on="click"}} />',
+    '<button type="button" {{action this.handleMouseover" on="mouseOver"}} />',
+    '<button type="button" {{on "click" this.handleClick}} />',
+    '<button type="button" {{on "mouseover" this.handleMouseover}} />',
+
+    // valid buttons with "submit" type
+    '<button />',
+    '<button type="submit" />',
+    '<button type="submit" {{action this.handleMouseover" on="mouseOver"}} />',
+    '<button type="submit" {{on "mouseover" this.handleMouseover}} />',
+
+    // valid div elements
+    '<div/>',
+    '<div></div>',
+    '<div type="submit"></div>',
+    '<div type="submit" {{action this.handleClick"}}></div>',
+    '<div type="submit" {{on "click" this.handleClick}}></div>',
+  ],
+
+  bad: [
+    {
+      template: '<button {{action this.handleClick}} />',
+
+      result: {
```

Comment:
We have updated rules to use snapshot testing with `verifyResults()`. Mind switching to that?

Reply:
Sure, let me do it right now